pkg/collector/parser/receiver_generic.go

```diff
@@ -53,6 +53,16 @@ func (g *GenericReceiver) Ports() ([]corev1.ServicePort, error) {
 	}
 
 	if g.defaultPort > 0 {
+		if g.name == "zipkin" {
```

Comment:
I am not sure about this approach. The protocol type should be rather set in https://github.com/open-telemetry/opentelemetry-operator/blob/eb549cea898a53b84017cab05c6b51b3022df9aa/pkg/collector/parser/receiver_zipkin.go#L23 and then just used to configure the service

Reply:
@pavolloffay just to be clear for zipkin config we can expect something like this:
```yaml
zipkin:
---
zipkin:
  protocols:
    http:
      endpoint: "0.0.0.0:9999"
```

And in case of default config "zipkin:" is this ok to add to GenericReceiver fields like defaultProtocol and defaultAppProtocol?

Reply:
I think we will have to add a field there